web/src/components/NumberField.js

```diff
@@ -14,6 +14,9 @@ const NumberField = ({
 }) => {
   const [local, setLocal] = useState(`${value}`);
 
+  // rerender component when value changes
+  useEffect(() => setLocal(value), [value]);
+
```

Comment:
There's some stuff going on in this component that I don't quite understand. For instance, why is the `blurHandler` calling `onChange`? Why does the `changeHandler` need `useCallback`?

Reply:
I'm also confused by that. How would the `if (onChange)` be called inside the `blurHandler`

Reply:
I attempted to do some refactoring, removed `useCallback` and calling `onChange` from the `blurHandler`, and broke functionality. I'm going to leave this as-is, since it works.